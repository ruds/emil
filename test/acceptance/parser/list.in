[];
[1, 2, foo];
["bar", baz];
["bar"];
["bar",];
[,];
