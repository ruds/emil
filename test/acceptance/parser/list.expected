0001 (DeclTopDecl
         :decls
         ((ValDecl
             :bindings
             ((ValBind
                 :pat
                 (IdentifierPattern
                     :qualifiers
                     ()
                     :identifier
                     "6974"
                     :is_op
                     false
                     :is_prefix_op
                     false)
                 :expr
                 (ListExpr
                     :exprs
                     ())
                 :rec
                 false))
             :explicit_type_vars
             ())))
0002 (DeclTopDecl
         :decls
         ((ValDecl
             :bindings
             ((ValBind
                 :pat
                 (IdentifierPattern
                     :qualifiers
                     ()
                     :identifier
                     "6974"
                     :is_op
                     false
                     :is_prefix_op
                     false)
                 :expr
                 (ListExpr
                     :exprs
                     (
                         (BigintLiteralExpr
                             :val
                             1)
                         (BigintLiteralExpr
                             :val
                             2)
                         (IdentifierExpr
                             :qualifiers
                             ()
                             :identifier
                             "666F6F"
                             :is_op
                             false
                             :is_prefix_op
                             false)
                     ))
                 :rec
                 false))
             :explicit_type_vars
             ())))
0003 (DeclTopDecl
         :decls
         ((ValDecl
             :bindings
             ((ValBind
                 :pat
                 (IdentifierPattern
                     :qualifiers
                     ()
                     :identifier
                     "6974"
                     :is_op
                     false
                     :is_prefix_op
                     false)
                 :expr
                 (ListExpr
                     :exprs
                     (
                         (StringLiteralExpr
                             :val
                             "626172")
                         (IdentifierExpr
                             :qualifiers
                             ()
                             :identifier
                             "62617A"
                             :is_op
                             false
                             :is_prefix_op
                             false)
                     ))
                 :rec
                 false))
             :explicit_type_vars
             ())))
0004 (DeclTopDecl
         :decls
         ((ValDecl
             :bindings
             ((ValBind
                 :pat
                 (IdentifierPattern
                     :qualifiers
                     ()
                     :identifier
                     "6974"
                     :is_op
                     false
                     :is_prefix_op
                     false)
                 :expr
                 (ListExpr
                     :exprs
                     ((StringLiteralExpr
                         :val
                         "626172")))
                 :rec
                 false))
             :explicit_type_vars
             ())))
0005 ERROR
0006 ERROR
0007 (EndOfFileTopDecl)
